python/hsfs/core/feature_group_base_engine.py

```diff
@@ -152,7 +153,13 @@ def _verify_schema_compatibility(self, feature_group_features, dataframe_feature
             )
 
     def get_subject(self, feature_group):
+        _client = client.get_instance()
+
+        feature_store_id = None
+        if feature_group.feature_store.project_id != int(_client._project_id):
```

Comment:
This doesn't work. Reason being that the `feature_group.feature_store` object is Null - which was the original problem that made me investigate into this code path.

Reply:
when calling `get_feature_group` and so on it should be assigned.